src/Recurr/Transformer/InstanceGenerator.php

```diff
@@ -0,0 +1,770 @@
+     * @param int|null $iterationLimit An iteration limit that, when provided, will serve as a fail-safe
+     *   that stops the generator when the limit is reached to prevent an infinitely large iterator.
+     * @return Generator|DateTime[]
+     * @throws MissingData
+     */
+    public function generate(
+        DateTime $seed,
+        array $rrules = [],
+        array $rdates = [],
+        array $exrules = [],
+        array $exdates = [],
+        $ignoreCounts = false,
+        $iterationLimit = null
+    ) {
+        if (empty($rrules) && empty($rdates) && empty($exrules) && empty($exdates)) {
+
+            yield $seed; // Just yield the seed and nothing else, since there is no recurrence.
+
+            return; // Quit the generator.
+        }
+
+        // Use the timezone for the seed date.
+        $timezone = $seed->getTimezone();
+
+        // An array to hold all the generators for dates that we'll include.
+        $r_generators = [];
+
+        // An array to hold all the generators for dates that we'll exclude.
+        $ex_generators = [];
+
+        // Add a generator for the RDATE collection
+        $r_generators[] = $this->generateFromDates($rdates, $timezone);
+
+        // Add a generator for each RRULE
+        foreach ($rrules as $rrule) {
+            $cloned_rule = clone $rrule;
+            $cloned_rule->setStartDate($seed);
+            $r_generators[] = $this->generateFromRule(
+                $cloned_rule,
+                $timezone,
+                $ignoreCounts
+            );
+        }
+
+        // Add a generator for the EXDATE collection
+        $ex_generators[] = $this->generateFromDates($exdates, $timezone);
+
+        // Add a generator for each EXRULE
+        foreach ($exrules as $exrule) {
+            $cloned_rule = clone $exrule;
+            $cloned_rule->setStartDate($seed);
+            $ex_generators[] = $this->generateFromRule(
+                $cloned_rule,
+                $timezone,
+                $ignoreCounts
+            );
+        }
+
+        $iterations = 0;
+
+        do {
+            // Holds the next date provided by each `$r_generators`
+            $r_dates = [];
+
+            /** @type Generator $generator */
+            foreach ($r_generators as $key => $generator) {
+                if (!$generator->valid()) {
+                    // This generator has reached it's end... Unset if from our array and continue.
+                    unset($r_generators[$key]);
+
+                    continue;
+                }
+
+                $r_dates[] = $generator->current();
+            }
+
+            if (empty($r_dates)) {
+                // No next date could be found, so we're done here.
+                break;
+            }
+
+            // Get the lowest of all the `$r_dates` This is the "next" instance date for the series, so long as
+            // it's not also found in with the EXDATES or the EXRULEs.
+            $min_r_date = $this->minDate($r_dates);
```

Comment:
There seems to be quite a mix of **snake_case** and **camelCase** variable names here. Not exactly sure of the project's style, but it does seem to use camelCase conventionally.